pkg/install/quota.go

```diff
@@ -0,0 +1,83 @@
+package install
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+//checkQuota checks usage quotas vs. resources required by cluster before cluster creation
+func (i *Installer) checkQuota(ctx context.Context) error {
+	//prepare usage client
+	spp := &i.doc.OpenShiftCluster.Properties.ServicePrincipalProfile
+	r, err := azure.ParseResourceID(i.doc.OpenShiftCluster.ID)
+	if err != nil {
+		return err
+	}
+	conf := auth.NewClientCredentialsConfig(spp.ClientID, string(spp.ClientSecret), spp.TenantID)
+	authorizer, err := conf.Authorizer()
+	if err != nil {
+		return err
+	}
+	usageClient := compute.NewUsageClient(r.SubscriptionID, authorizer)
+	return i.calculateAndCompareQuota(ctx, usageClient)
+}
+
+func (i *Installer) calculateAndCompareQuota(ctx context.Context, uc compute.UsageClient) error {
+	//master nodes are already included in the declaration below
+	requiredResources := map[string]int{
+		"cores":              24,
+		"virtualMachines":    3,
+		"standardDSv3Family": 3,
+		"PremiumDiskCount":   3,
+	}
+	if api.VMSizeStandardD8sV3 != i.doc.OpenShiftCluster.Properties.MasterProfile.VMSize {
```

Comment:
Should this not be in the validation? catch early?

Reply:
@Makdaam when you make a freestanding function below, use it for 3 * i.doc.OpenShiftCluster.Properties.MasterProfile.VMSize too.

Reply:
This was an artifact of a separate step to add master VM and worker VM resources.
The validity of VMSize is checked by api validation like @mjudeikis said. 
Added one function to add resources for all types of VMs.